realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1156,6 +1178,93 @@ public long count() {
     }
 
     /**
+     * Find all objects that fulfill the query conditions.
+     * Results will be posted to the callback instance {@link Realm.QueryCallback} asynchronously
+     *
+     * @return An {@link io.realm.RealmQuery.AsyncRequest} representing a cancellable, pending asynchronous query
+     * @see io.realm.RealmResults
+     * @throws java.lang.RuntimeException Any other error
+     */
+    public AsyncRequest findAll(Realm.QueryCallback<E> callback) {
+        // will use the Looper of the caller thread to post the result
+        final Handler handler = new EventHandler(callback);
+
+        // We need a pointer to the caller Realm, to be able to handover the result to it
+        final long callerSharedGroupNativePtr = realm.getSharedGroupPointer();
+
+        // Handover the query (to be used by a worker thread)
+        final long ptrQuery = query.handoverQuery(callerSharedGroupNativePtr);
+
+        // We need to use the same configuration to open a background SharedGroup (i.e Realm)
+        // to perform the query
+        final RealmConfiguration realmConfiguration = realm.getConfiguration();
+
+        // This call needs to be done on the caller's thread, since SG()->get_version_of_current_transaction is not thread safe
+        final long[] callerSharedGroupVersion = realm.getSharedGroupVersion();
+
+        pendingQuery = asyncQueryExecutor.submit(new Runnable() {
+            @Override
+            public void run() {
+                if (!Thread.currentThread().isInterrupted() && (null == asyncRequest || !asyncRequest.isCancelled())) {
+                    Realm bgRealm = null;
+
+                    try {
+                        //TODO Once SharedGroup is thread safe, start reusing a cached instance
+                        //     of a worker Realm to avoid the cost of opening/closing a SharedGroup
+                        //     for each query
+                        bgRealm = Realm.getInstance(realmConfiguration);
+
+                        // begin_read may throw an UnreachableVersionException if the provided version
+                        // is no longer available, we fail fast
+                        bgRealm.setSharedGroupAtVersion(callerSharedGroupVersion);
+
+                        // Run the query & handover the table view for the caller thread
+                        long tableViewPtr = query.findAllWithHandover(bgRealm.getSharedGroupPointer(), ptrQuery);
+
+                        if (IS_DEBUG && NB_ADVANCE_READ_SIMULATION-- > 0) {
```

Comment:
I'm open to suggestions :)
The goal here is to be able to generate `UnreachableVersionException` & test different retry policy